clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1933,11 +1941,81 @@ private Map<String, String> topicPartitionTags(TopicPartition tp) {
         }
     }
 
+    // Visible for testing
+    void maybeCloseFetchSessions(final Timer timer) {
+        final Cluster cluster = metadata.fetch();
+        final List<RequestFuture<ClientResponse>> requestFutures = new ArrayList<>();
+        for (final Map.Entry<Integer, FetchSessionHandler> entry : sessionHandlers.entrySet()) {
+            final FetchSessionHandler sessionHandler = entry.getValue();
+            // set the session handler to notify close. This will set the next metadata request to send close message.
+            sessionHandler.notifyClose();
+
+            final int sessionId = sessionHandler.sessionId();
+            final Integer fetchTargetNodeId = entry.getKey();
+            // FetchTargetNode may not be available as it may have disconnected the connection. In such cases, we will
+            // skip sending the close request.
+            final Node fetchTarget = cluster.nodeById(fetchTargetNodeId);
+            if (fetchTarget == null || client.isUnavailable(fetchTarget)) {
+                log.debug("Skip sending close session request to broker {} since it is not reachable", fetchTarget);
+                continue;
+            }
+
+            log.debug("Sending close request for fetch session: {} to node: {}", sessionId, fetchTarget);
```

Comment:
nit: I don't think this log is necessary, since we'll log in `sendFetchRequestToNode` later anyway. The output will looks duplicated like this:

```
Sending close request for fetch session: 1234 to node: 0
Sending request(...) uncommitted to broker 0
```

Reply:
removed in latest commit